### DISCUSSION TITLE
RocksDb Kafka Table over Docker

### DISCUSSION THREAD

**Discussion by thiagorenangomes** [LGouellec/streamiz#263] [Q&A] [answered]
Hello everyone,

I'm facing a problem when try running a Global Kafka Table over a docker container. 

`builder.GlobalTable("info", RocksDb.As<string, string, StringSerDes, StringSerDes>($"infostore"));`

The code is very simple and works great on a console app using MemoryDB and RocksDB, but when I try run the equivalent on docker container using **RocksDB** I got an error:

> NativeImport.NativeLoadException: Unable to locate rocksdb native library, either install it, or use RocksDbNative nuget package

Is there any additional setup or is it a real limitation?  

![image](https://github.com/LGouellec/kafka-streams-dotnet/assets/28765453/446f0ba5-fc03-43ee-b0af-ffb3bc39d1ec)
![image](https://github.com/LGouellec/kafka-streams-dotnet/assets/28765453/e4a5d518-ef4d-49ca-835a-4be782766705)

>  /app/bin/Debug/netcoreapp3.1/runtimes/linux-x64/native/librocksdb-7.7.3.so: (NativeLoadException) dlopen: Error loading shared library /app/bin/Debug/netcoreapp3.1/runtimes/linux-x64/native/librocksdb-7.7.3.so: No such file or directory
      /app/bin/Debug/netcoreapp3.1/runtimes/linux-x64/native/librocksdb-7.7.so: (NativeLoadException) dlopen: Error loading shared library /app/bin/Debug/netcoreapp3.1/runtimes/linux-x64/native/librocksdb-7.7.so: No such file or directory
      /app/bin/Debug/netcoreapp3.1/runtimes/linux-x64/native/librocksdb-7.so: (NativeLoadException) dlopen: Error loading shared library /app/bin/Debug/netcoreapp3.1/runtimes/linux-x64/native/librocksdb-7.so: No such file or directory
      /app/bin/Debug/netcoreapp3.1/runtimes/linux-x64/native/librocksdb.so: (NativeLoadException) dlopen: Error loading shared library libsnappy.so.1: No such file or directory (needed by /app/bin/Debug/netcoreapp3.1/runtimes/linux-x64/native/librocksdb.so)

> **/app/bin/Debug/netcoreapp3.1/runtimes/linux-x64/native/librocksdb.so: (NativeLoadException) dlopen: Error loading shared library libsnappy.so.1: No such file or directory (needed by /app/bin/Debug/netcoreapp3.1/runtimes/linux-x64/native/librocksdb.so)**

**Comment by LGouellec** [comment, marked as answer]
Hi @thiagorenangomes, 

You have to install some dependencies at the container level. 
Please have a look [here](https://github.com/LGouellec/kafka-streams-dotnet-samples/blob/b417e7bfa924d121f4ba06362eded1455c261537/src/global-ktable/Dockerfile#L18).

Best regra
